### PR TITLE
fix: @babel/eslint-parser & upgrade prettier configs

### PR DESCRIPTION
**Pull request by xiaofan2406**
- `babel-eslint` -> `@babel/eslint-parser`
   Still need the babel parser due the the static class properties usage.
- `prettier` upgrade & config changes
- all other deps upgrade

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -29,6 +27,11 @@ module.exports = {
     react: {
       version: 'detect',
     },
+    'import/resolver': {
+      node: {
+        extensions: ['.ts', '.tsx', '.js', '.jsx'],
+      },
+    },
```
Comment (xiaofan2406) [inline index.js:34]: add this in since we by default support ts

**Review by pphminions — DISMISSED** [review]
LGTM 📟

**Comment by knilink** [comment]
maybe you can add `no-exclusive-tests`, `no-skipped-tests` as well :trollface: 
https://adslot.atlassian.net/browse/ADS-4393

**Review by knilink — APPROVED** [review]

**Review by vinteo — APPROVED** [review]

**Review by bchew — APPROVED** [review]
Merging :rocket:

**Pull request merged**